app/src/test/java/com/github/bsideup/liiklus/GroupVersionTest.java

```diff
@@ -0,0 +1,117 @@
+package com.github.bsideup.liiklus;
+
+import com.github.bsideup.liiklus.protocol.*;
+import com.github.bsideup.liiklus.protocol.ReceiveReply.Record;
+import com.github.bsideup.liiklus.test.AbstractIntegrationTest;
+import com.google.protobuf.ByteString;
+import lombok.val;
+import org.junit.Before;
+import org.junit.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+public class GroupVersionTest extends AbstractIntegrationTest {
+
+    private String topic;
+
+    @Before
+    public void setUpGroupVersionTest() throws Exception {
+        topic = testName.getMethodName();
+
+        // Will create a topic and initialize every partition
+        Flux.fromIterable(PARTITION_UNIQUE_KEYS)
+                .flatMap(key -> Mono
+                        .defer(() -> stub
+                                .publish(
+                                        PublishRequest.newBuilder()
+                                                .setTopic(topic)
+                                                .setKey(ByteString.copyFromUtf8(key))
+                                                .setValue(ByteString.copyFromUtf8(UUID.randomUUID().toString()))
+                                                .build()
+                                )
+                        )
+                        .repeat(10)
+                )
+                .blockLast();
+    }
+
+    @Test
+    public void testCompoundGroupId() throws Exception {
+        val groupVersion = 1;
+
+        val committedOffset = 3;
+        val nextOffset = committedOffset + 1;
+
+        ackOffset(groupVersion, committedOffset);
+
+        assertThat(getRecords(Optional.of(groupVersion)).blockFirst(Duration.ofSeconds(10)).getOffset())
+                .isEqualTo(nextOffset);
+
+        // Old style should also work
+        assertThat(getRecords(Optional.empty()).blockFirst(Duration.ofSeconds(10)).getOffset())
+                .isEqualTo(nextOffset);
+    }
+
+    @Test
```

Comment:
should we also add check when a higher version 'ack' and lower version tries to getRecords? I think version will not be decreasing, but just to add.